src/index.html

```diff
@@ -7,6 +7,38 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <div class="calendar calendar--month-length--28 calendar--start-day--wed">
+      <div class="calendar__day calendar__day--1"></div>
+      <div class="calendar__day calendar__day--2"></div>
+      <div class="calendar__day calendar__day--3"></div>
+      <div class="calendar__day calendar__day--4"></div>
```

Comment:
I don't think you need such modifiers  as `calendar__day--1`, you can set styles for all `calendar__day` using 
`@for $i from 1 through 31 {
      &:nth-child(n + #{$i}):before {
        content: "#{$i}";
      }
    }`